### DISCUSSION TITLE
Serverless redash

### DISCUSSION THREAD

**Discussion by teambob** [getredash/redash#6023] [Ideas]
It would be good to have a serverless version of redash which runs on AWS Lambda or GCP Cloud Functions, so there is no need to patch a server and have a server running 24/7

**Comment by justinclift** [comment]
There's no real chance of that happening, perhaps ever.

The architecture of Redash doesn't work in a way that would fit with that, and doesn't really seem like the kind of thing where it could.

**Comment by rshedzer** [comment]
I wonder if this might actually be possible. Redash does allow multiple instances no? assuming you host postgres and redis separately. Would it be theoretically possible to use [AWS Lambda Adapter](https://github.com/awslabs/aws-lambda-web-adapter) to serve traffic to people in a Redash lambda and a small instance to run cron jobs?

**Reply by justinclift** [reply]
Hmmm, thinking that through... you *might* be able to get something basic working, then build it out from there.  Interesting... :smile:

**Reply by adrianoesch** [reply]
Hi, I'm also curious about a serverless deployment option. I'm a big fan of putting flask services on lambda (using [awsgi](https://github.com/slank/awsgi)) storing states in redis. I've been deploying [dash](https://dash.plotly.com/) apps like this for a few years now. So I would think that postgres, redis and server should be fine, but what about the worker? How do the server and worker communicate? Does the worker require long running processes?

**Reply by teambob** [reply]
AWS EventBridge can be used to initiate scheduled tasks
